testing/testcases/gitplugin/cases/project.go

```diff
@@ -0,0 +1,328 @@
+				})
+			})
+			It("should return the created project", func() {
+				Expect(err).Should(BeNil())
+				Expect(projectList).ShouldNot(BeNil())
+				Expect(projectList.Items).ShouldNot(BeEmpty())
+				gotProject := FindByName(ToPtrList(projectList.Items), wantProject.GetName())
+				checkProjectRequiredFields(gotProject, &wantProject)
+			})
+		})
+	})
+})
+
+var caseGetProject = P0Case("test for getting project info").
+	WithCondition(
+		PluginImplementCondition{Interface: (*client.ProjectGetter)(nil)},
+	).WithFunc(func(testContext *TestContext) {
+	var (
+		ctx         context.Context
+		instance    TestablePlugin
+		projectName *string
+		projectInfo *v1alpha1.Project
+		err         error
+	)
+
+	BeforeEach(func() {
+		ctx = testContext.Context
+		instance = GitPluginFromCtx(ctx)
+		projectName = pointer.String("")
+		ctx = context.WithValue(ctx, v1alpha1.KeyForSubType, string(v1alpha1.GitUserProjectSubType))
+		projectInfo = &v1alpha1.Project{}
+		err = nil
+	})
+
+	JustBeforeEach(func() {
+		getter := instance.(client.ProjectGetter)
+		projectInfo, err = getter.GetProject(ctx, *projectName)
+	})
+
+	Context("project not exist", func() {
+		BeforeEach(func() {
+			projectName = pointer.String("not-exist-not-exist-not-exist")
+		})
+		It("get return 404", func() {
+			Expect(errors.IsNotFound(err)).Should(BeTrue())
+		})
+	})
+
+	Context("project exist", func() {
+		BeforeEach(func() {
+			projectName = pointer.String(instance.GetTestOrgProject())
+		})
+
+		It("should return the project", func() {
+			Expect(err).Should(BeNil())
+			Expect(projectInfo).ShouldNot(BeNil())
+			Expect(projectInfo.GetName()).Should(Equal(instance.GetTestOrgProject())) // TODO: more asserts
+		})
+	})
+})
+
+var caseCreateProject = P0Case("test for creating new project").
+	WithCondition(
+		PluginImplementCondition{Interface: (*client.ProjectCreator)(nil)},
+	).WithFunc(func(testContext *TestContext) {
+	var (
+		ctx         context.Context
+		instance    TestablePlugin
+		project     *v1alpha1.Project
+		err         error
+		projectName string
+		creator     client.ProjectCreator
+	)
+
+	BeforeAll(func() {
+		projectName = "create-project-" + rand.String(4)
+	})
+
+	BeforeEach(func() {
+		ctx = testContext.Context
+		instance = GitPluginFromCtx(ctx)
+		ctx = context.WithValue(ctx, v1alpha1.KeyForSubType, string(v1alpha1.GitUserProjectSubType))
+		project = &v1alpha1.Project{}
+		err = nil
+		creator = instance.(client.ProjectCreator)
+	})
+
+	JustBeforeEach(func() {
+		project.Name = projectName
+		project, err = creator.CreateProject(ctx, project)
+	})
+
+	It("project created successfully", func() {
+		Expect(err).Should(BeNil())
+		Expect(project).ShouldNot(BeNil())
+		Expect(project.GetName()).Should(Equal(projectName))
+		checkProjectRequired(project)
+	})
+
+	It("project exist, create failed", func() {
+		Expect(err).ShouldNot(BeNil())
+		Expect(project).Should(BeNil())
+	})
+
+	AfterAll(func() {
+		cleanupProject(ctx, instance, projectName)
+	})
```

Comment:
还是要保持一定的结构，直接It 放在最外层有点奇怪
```suggestion
    AfterAll(func() {
		cleanupProject(ctx, instance, projectName)
	})
    When("first create project", func() {
        It("should create successfully", func() {
            project, err = creator.CreateProject(ctx, project)

            Expect(err).Should(BeNil())
            Expect(project).ShouldNot(BeNil())
            Expect(project.GetName()).Should(Equal(projectName))
            checkProjectRequired(project)
        })
    })
     When("create an existed project", func() {
        It("should create failed", func() {
            project, err = creator.CreateProject(ctx, project)

            Expect(err).ShouldNot(BeNil())
		    Expect(project).Should(BeNil())
        })
    })
```